src/main/java/org/lscode/DupLocator/FileStorage.java

```diff
@@ -0,0 +1,94 @@
+/**
+ *  file:    FileStorage.java
+ *  desc:    concrete class for storing FileArray objects segregated according to a chosen criterion
+ *  author:  lscode
+ *  license: GNU General Public License v3.0
+ */
+
+package org.lscode.DupLocator;
+
+import java.util.*;
+
+public class FileStorage<T> implements Iterable<FileArray>{
+
+    protected Map<T, FileArray> fileMap = new HashMap<>();
+
+//    public FileStorage(){}
+
+//    public FileStorage(FileStorage<T> source){
+//    }
+
+    public Iterator<FileArray> iterator() {
+        return fileMap.values().iterator();
+    }
+
+    public void put(T criterion, FileData fileData){
+        FileArray fileList;
+
+        if (fileMap.containsKey(criterion)){
+            fileList = fileMap.get(criterion);
+        }
+        else{
+            fileList = new FileArray();
+            fileMap.put(criterion, fileList);
+        }
+        fileList.add(fileData);
```

Comment:
domyslam sie ze po to... tak sie nie robi. Ten element w fileMap bedzie mial dodany fileData w liscie(jezeli zostanie wykonana linia 33) ale i tak to oznacza piekło dla przyszlego Ciebie który bedzie musial np za rok poprawic jakiegos buga zwiazanego z fileMap